### DISCUSSION TITLE
Natasha 编译 完整代码的错误提示，有办法明确一些吗？

### DISCUSSION THREAD

**Discussion by daonidedie** [dotnetcore/Natasha#212] [Q&A]
想弄一个在线生成sln 的在线运行环境，准备使用natasha 但发现错误提示不太明确。这个如果需要改善应该怎么弄呢。
如，NatashaException: ; expected at AssemblyCSharpBuilder.Add(String script) at  xxx.xxx 

实际它不定位到多少行。这个是需要先进行语法分析吗？还是怎么弄，希望作者能够给出建议，。

**Comment by NMSAzulX** [comment]
1. 自己处理错误可以参考:
 - https://github.com/dotnetcore/Natasha/blob/main/test/ut/NatashaFunctionUT/Compile/CompileLogTest.cs#L25
 - https://github.com/dotnetcore/Natasha/blob/main/src/Natasha.CSharp/Natasha.CSharp/Public/Component/Exception/NatashaExceptionAnalyzer.cs
2. 如果使用 Natasha 的异常,可以:
```c#
try{
  .....
}catch(NatashaException e){
    throw new Exception(e.Message +"\r\n 源代码 \r\n" + e.Formatter);
}
```

希望可以帮助到你.
